dask_expr/tests/test_collection.py

```diff
@@ -2349,3 +2349,38 @@ def test_filter_optimize_condition():
 def test_scalar_repr(df):
     result = repr(df.size)
     assert result == "<dask_expr.expr.Scalar: expr=df.size(), dtype=int64>"
+
+
+def test_reset_index_filter_pushdown(df):
+    q = df.reset_index()
+    result = q[q.x > 5]
+    expected = df[df.x > 5].reset_index()
```

Comment:
Hm good point, I think I am ok with this since the index is meaningless anyway after reset_index (every partition starts at 0 compared to pandas where the index will be monotonic afterwards)

Reply:
Works for me. FWIW, I think we may want to formalize our approach to (not) maintaining indices at some point.